atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/TableMetadataPersistenceTest.java

```diff
@@ -22,56 +22,47 @@
-import org.junit.runners.Parameterized;
-import org.junit.runners.Parameterized.Parameters;
-
-/* TODO(boyoruk): Migrate to JUnit5 */
-@RunWith(Parameterized.class)
+import java.util.List;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
 @SuppressWarnings("checkstyle:all") // too many warnings to fix
 public class TableMetadataPersistenceTest {
 
     private static final int CUSTOM_COMPRESSION_BLOCK_SIZE = 32;
     private static final int STREAM_STORE_BLOCK_SIZE_WITH_COMPRESS_IN_DB = 256;
     private static final int UNSET_BLOCK_SIZE = 0;
 
-    private final TableDefinition tableDefinition;
-    private final int compressionBlockSizeKB;
-
-    @Parameters
-    public static Collection<Object[]> testCases() {
-        Collection<Object[]> params = new ArrayList<>();
+    public static List<Arguments> getParameters() {
+        List<Arguments> params = new ArrayList<>();
 
-        params.add(new Object[] {getRangeScanWithoutCompression(), UNSET_BLOCK_SIZE});
-        params.add(new Object[] {getDefaultExplicit(), AtlasDbConstants.DEFAULT_TABLE_COMPRESSION_BLOCK_SIZE_KB});
-        params.add(new Object[] {
-            getDefaultRangeScanExplicit(), AtlasDbConstants.DEFAULT_TABLE_WITH_RANGESCANS_COMPRESSION_BLOCK_SIZE_KB
-        });
-        params.add(new Object[] {getCustomExplicitCompression(), CUSTOM_COMPRESSION_BLOCK_SIZE});
-        params.add(new Object[] {getCustomTable(), CUSTOM_COMPRESSION_BLOCK_SIZE});
-        params.add(new Object[] {getStreamStoreTableWithCompressInDb(), STREAM_STORE_BLOCK_SIZE_WITH_COMPRESS_IN_DB});
-        params.add(new Object[] {getStreamStoreTableDefault(), UNSET_BLOCK_SIZE});
+        params.add(Arguments.of(getRangeScanWithoutCompression(), UNSET_BLOCK_SIZE));
+        params.add(Arguments.of(getDefaultExplicit(), AtlasDbConstants.DEFAULT_TABLE_COMPRESSION_BLOCK_SIZE_KB));
+        params.add(Arguments.of(
+                getDefaultRangeScanExplicit(),
+                AtlasDbConstants.DEFAULT_TABLE_WITH_RANGESCANS_COMPRESSION_BLOCK_SIZE_KB));
+        params.add(Arguments.of(getCustomExplicitCompression(), CUSTOM_COMPRESSION_BLOCK_SIZE));
+        params.add(Arguments.of(getCustomTable(), CUSTOM_COMPRESSION_BLOCK_SIZE));
+        params.add(Arguments.of(getStreamStoreTableWithCompressInDb(), STREAM_STORE_BLOCK_SIZE_WITH_COMPRESS_IN_DB));
+        params.add(Arguments.of(getStreamStoreTableDefault(), UNSET_BLOCK_SIZE));
 
         return params;
     }
 
-    public TableMetadataPersistenceTest(TableDefinition tableDefinition, int compressionBlockSizeKB) {
-        this.tableDefinition = tableDefinition;
-        this.compressionBlockSizeKB = compressionBlockSizeKB;
-    }
-
-    @Test
-    public void testSerializeAndDeserialize() {
+    @ParameterizedTest
+    @MethodSource("getParameters")
+    public void testSerializeAndDeserialize(TableDefinition tableDefinition) {
```

Comment:
How does it work when you have more parameters in the `getParameters` than you function takes. Does it just ignores the remaining ones?

Reply:
Exactly. We don't use the remaining params, so it makes sense not to take them.